l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol

```diff
@@ -59,6 +59,24 @@ library DynamicIncrementalMerkle {
         return bytes32(0);
     }
 
+    /**
+     * @dev Resets the tree to a blank state.
+     * Calling this function on MerkleTree that was already setup and used will reset it to a blank state.
+     */
+    function reset(Bytes32PushTree storage self, bytes32 zero) internal returns (bytes32 initialRoot) {
+        self._nextLeafIndex = 0;
+        for (uint256 i = self._zeros.length; 0 < i; --i) {
```

Comment:
you can cache `self._zeros.length`

Reply:
Does that need to be cached? I thought only the check and the decrement are called multiple times?